alpha-blog/app/views/categories/new.html.erb

```diff
@@ -0,0 +1,24 @@
+<h1 class="text-center mt-4">Create a new category</h1>
+<div class="container">
+  <div class="row justify-content-center">
+    <div class="col-10">
+      <%= render 'shared/errors', data: @category %>
+      <%= form_with(model: @category, class: "shadow p-3 mb-3 bg-dark rounded", local: true) do |f| %>
+        <div class="form-group row mt-1">
+          <%= f.label :name, "Category name", class: "col-2 col-form-label text-light" %>
+          <div class="col-10">
+            <%= f.text_field :name, class: "form-control shadow rounded", placeholder: "Enter a name" %>
```

Comment:
Shouldn't you add `required` here since this is required?

Reply:
But then in models already has `presence: true`. If the user did not enter anything, there will be an error
![image](https://user-images.githubusercontent.com/67599130/121564289-df257a80-ca4d-11eb-862e-7f78d6e1f962.png)
Double validation better? I have to add to other forms as well?

Reply:
There is 2 kinds of validation.
1. server side
2. frontend side.

What I am asking you to do is [this](https://www.w3schools.com/tags/att_input_required.asp).
If you don't need to hit the server, then don't. Imagine if all user hit the server just to get back an error like this which is easily set in the frontend, use up server resources to call the db, try to create record, run validation, fail, return data to frontend. Vs, frontend check, fail, feedback. kaotim

Reply:
ooo okok noted will edit👍